### ISSUE TITLE
Update should be using git data instead of file stat

### ISSUE THREAD

**Issue by radeksimko** [mjibson/party#17]
1. `go get -u github.com/Sirupsen/logrus`
2. `go get github.com/fsouza/go-dockerclient && cd $GOPATH/src/github.com/fsouza/go-dockerclient`
3. `party -d external -c -u`

Expected outcome of the above steps for me was to see the latest version of `github.com/Sirupsen/logrus` copied over from `$GOPATH/src/github.com/Sirupsen/logrus` to `external/github.com/Sirupsen/logrus` inside `$GOPATH/src/github.com/fsouza/go-dockerclient`.

Because party [uses fs data](https://github.com/mjibson/party/blob/master/main.go#L207) to find out what files to update and fresh clone/go-get of the parent project (`github.com/fsouza/go-dockerclient`) including its dependencies was considered newer (because of the sequence of steps I described above), no update was made.

I believe that party would benefit from using git data and date/time from git log instead of relying on file stats.

You may argue that I should have just used `party` to do the update instead of running `go get -u github.com/Sirupsen/logrus` outside of party, but when managing multiple go projects, each having different dependency management solution, I would not expect user to do it.

**Comment by mjibson** [comment]
Yes, this is true. I don't use party myself anymore (I've switched to govendor), so I'm unlikely to fix this.
